### PR TITLE
Fix generation callback interval

### PR DESCRIPTION
**Pull request by dakinggg**

### CODE REVIEW

File: examples/common/generate_callback.py [hunk 1]
```diff
@@ -43,7 +45,14 @@ def init(self, state: State, logger: Logger):
                 if isinstance(destination, WandBLogger):
                     self.wandb_logger = destination
 
-    def eval_end(self, state: State, logger: Logger):
+    def batch_checkpoint(self, state: State, logger: Logger):
+        if (state.timestamp.batch.value % self.batch_log_interval) == 0:
+            self.generate(state, logger)
+
+    def fit_end(self, state: State, logger: Logger):
+        self.generate(state, logger)
```
Comment (vchiley) [inline examples/common/generate_callback.py:53]:
```suggestion
        if (state.timestamp.batch.value % self.batch_log_interval) != 0:
            self.generate(state, logger)
```

prevents 2x generates at `fit_end`.

**Review by vchiley — APPROVED** [review]
lgtm

would adding a default batch_log_interval which only generates at the end of training make sense?

File: examples/common/generate_callback.py (hunk 1 above)
Comment (vchiley) [reply examples/common/generate_callback.py:53]: feel free to ignore if it doesn't make sense.

Comment (dakinggg) [reply examples/common/generate_callback.py:53]: nah, makes sense

**Review by vchiley — APPROVED** [review]
lgtm

**Review by vchiley — APPROVED** [review]

**Pull request merged**